client/state/jetpack-connect/actions.js

```diff
@@ -361,10 +361,27 @@ export function authorize( queryObject ) {
 					data: data,
 					error: null,
 				} );
+
 				// Update the user now that we are fully connected.
 				const user = userFactory();
 				user.fetching = false;
 				user.fetch();
+
+				// @TODO: When user fetching is reduxified, let's get rid of this hack.
+				// Currently, we need it to make sure user has been refetched before we continue.
+				// Otherwise the user might see a confusing message that they have no sites.
+				// See p8oabR-j3-p2/#comment-2399 for more information.
+				return new Promise( resolve => {
+					const userFetched = setInterval( () => {
+						const loadedUser = user.get();
```

Comment:
What will happen if getting user fails — can `loadedUser` end up being falsy?

Reply:
The user will see the same error that was reported. That's how it works currently, though, too, so it's not a regression.

I agree - this is not ideal and still can be improved. But I'd love to not have to layer hacks on top of hacks. Ideally, we should get rid of those hacks once `lib/user` is reduxified.